### PR TITLE
Update KubeOne dependency and image to latest version

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
This PR updates the go module and image for KubeOne to use the changes from latest commit https://github.com/kubermatic/kubeone/commit/0536063ab064601ba217c2abd41abd4c80a02477.

**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

/kind chore

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -351,3 +349,9 @@ require (
 )
 
 require github.com/jackpal/gateway v1.0.10
+
+require (
+	github.com/ajeddeloh/yaml v0.0.0-20170912190910-6b94386aeefd // indirect
+	github.com/coreos/container-linux-config-transpiler v0.9.0 // indirect
+	github.com/coreos/ignition v0.35.0 // indirect
+)
```
Comment (xrstf) [inline go.mod]: Please merge this into the existing require() blocks for indirect dependencies.

Comment (ahmedwaleedmalik) [reply go.mod]: Done

Comment (ahmedwaleedmalik) [reply go.mod]: Done

**Review by xrstf — APPROVED** [review]
/approve

**Comment by ahmedwaleedmalik** [comment]
cc @kubermatic/sig-release-managers

File: go.mod [hunk 2]
```diff
@@ -127,6 +127,8 @@ replace (
 
 replace github.com/ajeddeloh/go-json => github.com/coreos/go-json v0.0.0-20220810161552-7cce03887f34
 
+replace github.com/kubermatic/machine-controller => github.com/kubermatic/machine-controller v1.56.1-0.20230601122944-bfd2fd7cf541
```
Comment (xrstf) [inline go.mod:130]: Why do we need this explicit replace directive now?

Comment (ahmedwaleedmalik) [reply go.mod:130]: On updating the `kubeone` package and running `go mod tidy` the MC version gets bumped to v1.56.2. Ideally should be > v1.56.1-0.20230601122944-bfd2fd7cf541(latest commit) but it isn't.

Comment (embik) [reply go.mod:130]: As much as `replace` is a quick fix for this, do we have any idea why `go mod` is starting to resolve this differently? Afaik the previous KubeOne version also has a dependency on machine-controller, so I think we should try to figure out why this is happening?

Comment (ahmedwaleedmalik) [reply go.mod:130]: Sure you can create a separate ticket if you want to investigate. Right now all I want is the latest version of KubeOne to test a few things with external clusters; which this PR should do.

Comment (xrstf) [reply go.mod:130]: Please add at least a comment explaining why we are _seemingly_ downgrading the MC with a replace directive.

> Sure you can create a separate ticket if you want to investigate.

:zipper_mouth_face:

**Comment by xrstf** [comment]
/approve
/lgtm

🤐

File: go.mod (hunk 2 above)
Comment (embik) [reply go.mod:130]: > Sure you can create a separate ticket if you want to investigate. Right now all I want is the latest version of KubeOne to test a few things with external clusters; which this PR should do.

I'm sorry but this is not someone else's responsibility to track. You want to put a hack in place, that's okay temporarily, but it's your hack, so please track it accordingly.

Comment (ahmedwaleedmalik) [reply go.mod:130]: You are the one bothered about it and calling it a "hack".

**Pull request merged**